src/storage/buffer_manager/buffer_manager.cpp

```diff
@@ -193,25 +199,35 @@ void handleAccessViolation(unsigned int exceptionCode, PEXCEPTION_POINTERS excep
 
 // Returns true if the function completes successfully
 inline bool try_func(const std::function<void(uint8_t*)>& func, uint8_t* frame,
-    const std::vector<std::unique_ptr<VMRegion>>& vmRegions, PageSizeClass pageSizeClass) {
-#if defined(_WIN32)
+    const std::array<std::unique_ptr<VMRegion>, 2>& vmRegions [[maybe_unused]],
+    PageSizeClass pageSizeClass [[maybe_unused]]) {
+#if BM_MALLOC
+    if (frame == nullptr) {
+        return false;
+    }
+#endif
+
+#if defined(_WIN32) && !BM_MALLOC
     try {
 #endif
         func(frame);
-#if defined(_WIN32)
+#if defined(_WIN32) && !BM_MALLOC
     } catch (AccessViolation& exc) {
         // If we encounter an acess violation within the VM region,
         // the page was decomitted by another thread
         // and is no longer valid memory
+#if BM_MALLOC
```

Comment:
How can this be true when `#if defined(_WIN32) && !BM_MALLOC` is true?

Reply:
I think I added those at two different times.

I'm not sure if accessing freed memory may cause an access violation on Windows, in which case optimisticRead won't be safe even when using malloc unless we handle those access violations (at least, those that occur within the frame).
The CI is passing with it disabled at the moment, which makes me wonder if it's not covered by tests, or if it in practice just isn't happening (maybe it doesn't actually happen at all, or it's just that freed memory that's been re-used wouldn't cause the access violation).
I think I'll just remove this block for now and leave the access violation handling disabled with `BM_MALLOC`.